### PR TITLE
Kafka streams  default config fix

### PR DESCRIPTION
**Pull request by laxmanchekka**

### CODE REVIEW

File: kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java [hunk 1]
```diff
@@ -151,15 +158,36 @@ public boolean healthCheck() {
...
+    // default = 16384
+    baseStreamsConfig.put(producerPrefix(BATCH_SIZE_CONFIG), "524288");
+    // Enable compression on producer for better throughput
+    // default - none
     baseStreamsConfig.put(producerPrefix(COMPRESSION_TYPE_CONFIG), CompressionType.GZIP.name);
-    baseStreamsConfig.put(producerPrefix(MAX_REQUEST_SIZE_CONFIG), "10485760");
+    // Allows to send larger messages to broker
+    // Requires tuning on broker, topic and consumer application as well
+    // default = 1048576
+    baseStreamsConfig.put(producerPrefix(MAX_REQUEST_SIZE_CONFIG), "1048576");
```
Comment (kotharironak) [inline kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java:176]: Default and new values seem the same. Did we miss `0` at the end of` 10485760`?

File: kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java [hunk 2]
```diff
@@ -151,15 +158,36 @@ public boolean healthCheck() {
...
 
     // Default consumer configurations
+    // Increase to enable receiving large messages
+    // default = 1048576
+    baseStreamsConfig.put(consumerPrefix(MAX_PARTITION_FETCH_BYTES_CONFIG), "1048576");
```
Comment (kotharironak) [inline kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java:181]: Here as well default and set value is same - `1048576`

**Review by kotharironak — DISMISSED** [review]
Minor comments to bring to your notice about default values. Rest looks fine.

File: kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java (hunk 1 above)
Comment (laxmanchekka) [reply kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java:176]: I will remove this config from as this is default too.
This is not a standalone configuration and requires tuning few other producer, broker, consumer configs together.

File: kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java (hunk 2 above)
Comment (laxmanchekka) [reply kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java:181]: Will remove this one too.

File: kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java [hunk 3]
```diff
@@ -1,13 +1,15 @@
 package org.hypertrace.core.kafkastreams.framework;
 
 
-import static org.apache.kafka.clients.consumer.ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG;
 import static org.apache.kafka.clients.consumer.ConsumerConfig.AUTO_OFFSET_RESET_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.MAX_PARTITION_FETCH_BYTES_CONFIG;
```
Comment (kotharironak) [inline kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java:5]:
```suggestion
```

File: kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java [hunk 4]
```diff
@@ -1,13 +1,15 @@
...
 import static org.apache.kafka.clients.consumer.ConsumerConfig.MAX_POLL_RECORDS_CONFIG;
```
Comment (kotharironak) [inline kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java:6]:
```suggestion
```

File: kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java [hunk 5]
```diff
@@ -1,13 +1,15 @@
...
+import static org.apache.kafka.clients.producer.ProducerConfig.ACKS_CONFIG;
 import static org.apache.kafka.clients.producer.ProducerConfig.BATCH_SIZE_CONFIG;
 import static org.apache.kafka.clients.producer.ProducerConfig.COMPRESSION_TYPE_CONFIG;
 import static org.apache.kafka.clients.producer.ProducerConfig.LINGER_MS_CONFIG;
 import static org.apache.kafka.clients.producer.ProducerConfig.MAX_REQUEST_SIZE_CONFIG;
```
Comment (kotharironak) [inline kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java:11]:
```suggestion
```

**Review by kotharironak — APPROVED** [review]

File: kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java (hunk 3 above)
Comment (kotharironak) [reply kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java:5]: Seems un-used.

File: kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java (hunk 5 above)
Comment (kotharironak) [reply kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java:11]: Seems un-used.

File: kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java (hunk 4 above)
Comment (kotharironak) [reply kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java:6]: Seems un-used.

**Review by kotharironak — APPROVED** [review]

**Pull request merged**